.github/workflows/golangci-lint.yaml

```diff
@@ -0,0 +1,43 @@
+---
+name: golangci-lint
+
+on:
+  pull_request:
+    branches: ['*']
+
+jobs:
+  golangci-lint:
+    name: golangci-lint
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set GO_VERSION
+        run: echo "GO_VERSION=$(grep -m 1 -r 'go' go.mod | awk '{print $2}')" >> $GITHUB_ENV
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GO_VERSION }}
```

Comment:
These setup steps are repeating in multiple actions, can we refactor this out? I believe GitHub Actions has a construct for this